xapian-letor/m4-macros/xapianletor.m4

```diff
@@ -0,0 +1,172 @@
+# Get XAPIAN_CXXFLAGS, XAPIAN_LIBS, and XAPIAN_VERSION from xapianletor-config and
+# AC_SUBST() them.
+
+# serial 17
+
+# AC_PROVIDE_IFELSE(MACRO-NAME, IF-PROVIDED, IF-NOT-PROVIDED)
+# -----------------------------------------------------------
+# If this macro is not defined by Autoconf, define it here.
+m4_ifdef([AC_PROVIDE_IFELSE],
+	 [],
+	 [m4_define([AC_PROVIDE_IFELSE],
+		 [m4_ifdef([AC_PROVIDE_$1],
+			   [$2], [$3])])])
+
+# XO_LIB_XAPIANLETOR([ACTION-IF-FOUND [, ACTION-IF-NOT-FOUND[ ,XAPIANLETOR-CONFIG]]])
+# --------------------------------------------------------
+# AC_SUBST-s XAPIAN_CXXFLAGS, XAPIAN_LIBS, and XAPIAN_VERSION for use in
+# Makefile.am
+#
+# If ACTION-IF-FOUND and ACTION-IF-NOT-FOUND are both unset, then an
+# appropriate AC_MSG_ERROR is used as a default ACTION-IF-NOT-FOUND.
+# This allows XO_LIB_XAPIANLETOR to be used without any arguments in the
+# common case where Xapian is a requirement (rather than optional).
+#
+# XAPIANLETOR-CONFIG provides the default name for the xapianletor-config script
+# (which the user can override with "./configure XAPIANLETOR_CONFIG=/path/to/it").
+# If unset, the default is xapianletor-config.  Support for this third parameter
+# was added in Xapian 1.1.3.
+AC_DEFUN([XO_LIB_XAPIANLETOR],
+[
+  AC_ARG_VAR(XAPIANLETOR_CONFIG, [Location of xapianletor-config (default:] ifelse([$3], [], xapianletor-config, [$3]) [on PATH)])
+  dnl AC_PATH_PROG ignores an existing user setting of XAPIANLETOR_CONFIG unless it
+  dnl has a full path, so add special handling for such cases.
+  xapian_config_to_check_for="ifelse([$3], [], xapianletor-config, [$3])"
+  case $XAPIANLETOR_CONFIG in
+  "") ;;
+  */configure)
+    AC_MSG_ERROR([XAPIANLETOR_CONFIG should point to a xapianletor-config script, not a configure script.])
+    ;;
+  [[\\/]* | ?:[\\/]*)]
+    # XAPIANLETOR_CONFIG has an absolute path, so AC_PATH_PROG can handle it.
+    ;;
+  [*[\\/]?*)]
+    # Convert a relative path to an absolute one.
+    XAPIANLETOR_CONFIG=`pwd`/$XAPIANLETOR_CONFIG
+    ;;
+  *)
+    # If there's no path on XAPIANLETOR_CONFIG, use it as the name of the tool to
+    # search PATH for, so that things like this work:
+    #   ./configure XAPIANLETOR_CONFIG=xapianletor-config-1.3
+    xapian_config_to_check_for=$XAPIANLETOR_CONFIG
+    XAPIANLETOR_CONFIG=
+    ;;
+  esac
+  AC_PATH_PROG(XAPIANLETOR_CONFIG, "$xapian_config_to_check_for")
+  if test -z "$XAPIANLETOR_CONFIG"; then
+    ifelse([$2], ,
+      [ifelse([$1], , [
+	dnl Simple check to see if the problem is likely to
+	dnl be that we're using a "packaged" xapian-letor but
+	dnl only have the runtime package installed.
+	for sfx in '' 32 64 ; do
+	  set /usr/lib$sfx/libxapianletor.so.*
+	  if test "/usr/lib$sfx/libxapianletor.so.*" != "$1" ; then
+	    if test -r /etc/debian_version ; then
+	      pkg="libxapian-dev"
```

Comment:
There's no debian package for xapian-letor yet so there isn't a "right answer" here as such, but that's the package name for xapian-core's development libraries and headers, which is definitely a "wrong answer".

Reply:
fixed in a fixup commit.